src/lib.rs

```diff
@@ -40,23 +69,41 @@ impl Display for Error {
             Error::SpecifiedLengthTooLong(x) => write!(f, "SpecifiedLengthTooLong({x})"),
             Error::ProtoSerializationErr(x) => write!(f, "ProtoSerializationErr({x})"),
             Error::HeaderLengthOverflow(x) => write!(f, "HeaderLengthOverflow({x})"),
+            Error::EncryptError(x) => write!(f, "EncryptError({x})"),
+            Error::DecryptError(x) => write!(f, "DecryptError({x})"),
         }
     }
 }
 
-const PRE_HEADER_LEN: usize = 7;
-const MAGIC: &[u8; 4] = b"IRON";
-const V4: u8 = 4u8;
+/// These bytes are an attached payload, which means IV + CIPHERTEXT with no header.
```

Comment:
```suggestion
/// These bytes are an attached payload, which means IV + CIPHERTEXT without `0IRON` magic.
```

Reply:
The problem for me is that attached payloads by definition have _the_ edoc header attached. So there isn't "no header". There's "no direct header" or "no detached header" or "no magic header (except the one that'll be attached)". To me "without `0IRON` magic" in contrast to the detached comment on `EncryptedPayload` nearby is the most clear.

Reply:
Talked about it. Shortening to leave off the header bit.